src/zc/lockfile/__init__.py

```diff
@@ -83,15 +128,26 @@ def __init__(self, path):
             _lock_file(fp)
         except:
             fp.seek(1)
-            pid = fp.read().strip()[:20]
+            content = fp.read().strip()
+            # parse pid and/or hostname from the lock file
+            parse_template = LockFileParserFormatter().format(content_template)
+            match = re.search(parse_template, content)
+            if match:
+                values_str = ' '.join(
+                    '{0}={1}'.format(name, value[:20])
+                    for name, value in sorted(match.groupdict().items()))
+            else:
+                # lock file doesn't match the given format; just log what
+                # zc.lockfile originally did in similar situations
+                values_str = 'pid=UNKNOWN'
             fp.close()
-            if not pid:
-                pid = 'UNKNOWN'
-            logger.exception("Error locking file %s; pid=%s", path, pid)
+            logger.exception("Error locking file %s; %s", path, values_str)
             raise
 
+        # write pid and/or hostname into the lock file
+        content = LockFileContentFormatter().format(content_template)
```

Comment:
Why not just:

    content = content_template.format(pid=os.getpid(),
                                      hostname=socket.gethostname())